planner/core/integration_test.go

```diff
@@ -5899,8 +5905,10 @@ func TestIssue29221(t *testing.T) {
 	tk.MustQuery("explain format = 'brief' select /*+ use_index_merge(t) */ * from t where a = 1 or b = 1 limit 3;").Check(testkit.Rows(
```

Comment:
Do we have a test case with condition  a = 1 and b = 1, the IndexMerge type is not `OR`. I don't know whether we support AND indexmerge or not, but it is better to have the test case, and we shouldn't push limit down for that case.

Reply:
sure, let me add some